tools/options.c

```diff
@@ -366,7 +369,8 @@ static void nn_append_string_to_free (struct nn_parse_context *ctx,
     if (!lst->items) {
         nn_memory_error (ctx);
     }
-    lst->to_free[lst->to_free_num-1] = str;
+    nn_assert (lst->to_free);
```

Comment:
Look again -- it's `lst->to_free` that needs the assert because it's indexed with `[lst->to_free_num-1]`